src/raftstore/store/snap.rs

```diff
@@ -1985,9 +1986,17 @@ impl SnapManager {
                     error!("failed to parse snapkey from {}", name);
                     return None;
                 }
-                Some((SnapKey::new(numbers[0], numbers[1], numbers[2]), is_sending))
+                let snap_key = SnapKey::new(numbers[0], numbers[1], numbers[2]);
+                if core.registry.contains_key(&snap_key) {
```

Comment:
A snapshot is idle if it's not registered. But a snapshot will not be deleted once it's idle. During the snapshot GC, it checks the validation of the snapshot with the meta data such as term and indexes.

A snapshot generated and to be sent will be deleted only if its indexes lag behind the too much. In that case, sending a obsolete snapshot will not be necessary. A snapshot received but not applied yet would not be deleted, since it indexes would not lag behind. In both cases, the snapshot GC would delete snapshot if it's obsolete.